### PR TITLE
Added server-side 404 page for posts

### LINKED ISSUE #2785: Removed Posts should not be visible for non-moderator users
### Steps to Reproduce
Ideally, when items are removed, they should no longer be visible for anonymous and non-moderator users. Currently, while no longer in the channel and front pages, removed items can still be accessed directly from the URL. 

Steps:

1. Go to URL of item to remove. Copy the URL.
1. Remove the item.
1. Log out of Open or open another browser that isn't logged in.
1. Paste URL of removed item and access that page.
1. You can view the removed post.

### Expected Behavior
Ideally there'd be a 404 for non-moderator and anonymous users when they try to view removed items.

### Actual Behavior
Everyone can still view the removed items directly with the URL, even after removed.

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #2785 

#### What's this PR do?
This adds a server-side 404 page for removed posts for non-moderator users, to ensure that for search engines and anyone else don't see this content perpetually. Also covered 400/403 errors just in case since they were a quick win.

#### How should this be manually tested?
- Remove a post in a public and/or private channel
- You should still be able to view the post as a moderator
- As a logged in user you should get a 404 page
- As an anonymous user you should get a 404 page for the public post

#### Any background context you want to provide?

Intentionally not handled here:

- 500 error pages: not in scope and a bit of effort
- Anonymous users trying to view a removed private post

### CODE REVIEW

**Comment by pdpinch** [comment]
What effect will this have on the profile page's list of posts?

**Comment by rhysyngsun** [comment]
@pdpinch that wasn't addressed here, but the links to the posts will resolve to the 404

File: open_discussions/views.py [hunk 1]
```diff
@@ -102,8 +108,50 @@ def index(request, **kwargs):  # pylint: disable=unused-argument
...
+    return _render_app(request)
+
+
+def channel_post(request, **kwargs):
+    """Render a channel post's page as long as it isn't removed"""
+    post_id = kwargs.get("post_id", None)
+    if not post_id:
+        raise Http404("No post specified")
+
+    post = get_object_or_404(Post, post_id=post_id)
```
Comment (mbertrand) [inline open_discussions/views.py:125]: I noticed that the author of a removed post now gets a 404, but a removed comment is still reachable to the author via direct link.  Not sure if that really matters or not, all other non-admin users get a 404.

File: open_discussions/templates/base.html [hunk 2]
```diff
@@ -10,15 +10,11 @@
...
-    <script type="text/javascript">
-    var SETTINGS = {{ js_settings_json|safe }};
-    </script>
-    {% render_bundle 'style' %}
-    {% endif %}
+
     <title>{% block title %}{% trans "Open Discussions" %}{% endblock %}</title>
     <meta name="description" content="{% block description %}{% endblock %}">
     <meta name="keywords" content="{% block keywords %}{% endblock %}">
+    {% render_bundle 'style' %}
```
Comment (mbertrand) [inline open_discussions/templates/base.html:17]: On this branch but not master I'm seeing this error in the console:

Screen Shot 2020-07-30 at 10 13 45 AM

File: open_discussions/views.py (hunk 1 above)
Comment (mbertrand) [reply open_discussions/views.py:125]: To clarify, a removed comment on a non-removed post

File: docker-compose.override.yml [hunk 3]
```diff
@@ -5,7 +5,6 @@ services:
       - .:/src
       - django_media:/var/media
     environment:
-      DEBUG: 'True'
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply docker-compose.override.yml:8]: good catch, i made that change to test locally, but i like your fix

File: open_discussions/views.py (hunk 1 above)
Comment (rhysyngsun) [reply open_discussions/views.py:125]: Hm, are you ok with me opening a separate issue for that since it'll involve a bunch more test reworking?

Comment (mbertrand) [reply open_discussions/views.py:125]: Fine with me!

**Pull request merged**